pkgs/by-name/so/sourcegit/package.nix

```diff
@@ -0,0 +1,116 @@
+{
+  lib,
+  stdenv,
+  buildDotnetModule,
+  fetchFromGitHub,
+  dotnetCorePackages,
+  copyDesktopItems,
+  makeDesktopItem,
+  libicns,
+
+  libXcursor,
+  libXext,
+  libXi,
+  libXrandr,
+
+  git,
+  xdg-utils,
+}:
+
+buildDotnetModule (finalAttrs: {
+  pname = "sourcegit";
+  version = "2025.01";
+
+  src = fetchFromGitHub {
+    owner = "sourcegit-scm";
+    repo = "sourcegit";
+    tag = "v${finalAttrs.version}";
+    hash = "sha256-6FO0WRUcWm7xnuw6Br97+cWZhvzIOesg/eANZzeyxZo=";
+  };
+
+  patches = [ ./fix-darwin-git-path.patch ];
+
+  dotnet-sdk = dotnetCorePackages.sdk_9_0;
+  dotnet-runtime = dotnetCorePackages.runtime_9_0;
+
+  nugetDeps = ./deps.json;
+
+  projectFile = [ "src/SourceGit.csproj" ];
+
+  executables = [ "SourceGit" ];
+
+  dotnetFlags = [
+    "-p:DisableUpdateDetection=true"
+    "-p:DisableAOT=true"
+  ];
+
+  nativeBuildInputs = lib.optionals stdenv.hostPlatform.isLinux [
+    copyDesktopItems
+    libicns
+  ];
+
+  # these are dlopen-ed at runtime
+  # libXi is needed for right-click support
+  # not sure about what the other ones are needed for, but I'll include them anyways
+  runtimeDeps = [
+    libXcursor
+    libXext
+    libXi
+    libXrandr
+  ];
+
+  desktopItems = [
+    (makeDesktopItem {
+      name = "SourceGit";
+      exec = "SourceGit";
+      icon = "SourceGit";
+      desktopName = "SourceGit";
+      terminal = false;
+      comment = finalAttrs.meta.description;
+    })
+  ];
+
+  # add fallback binaries to use if the user doesn't already have them in their PATH
+  preInstall = ''
+    makeWrapperArgs+=(
+      --suffix PATH : ${
+        lib.makeBinPath [
+          git
+          xdg-utils
+        ]
```

Comment:
I'd prefer this use an overridable attr, e.g. `makeBinPath finalAttrs.runtimeInputs` (or `dependencies`, or something)?

This could make customization easier for end-users.

Reply:
Might be useful, especially since on darwin, the PATH isn't always inherited when running the program.